### PR TITLE
Format Python according to PEP 8

### PR DESCRIPTION
**Pull request by emileferreira**
I've formatted all the Python files in accordance with [PEP 8](https://peps.python.org/pep-0008/)—using space indentation, limiting lines to 79 characters, _et cetera_. This is in preparation for further, meaningful changes to the codebase. I've also removed unused imports and variables, except those in functions decorated as `UnfinishedFunc`.

### CODE REVIEW

**Comment by emileferreira** [comment]
I've also noticed some undefined variables in the formatted `FunctionalSupport.py` at lines 257 and 267, and in `Evaluator.py` at lines 227, 504 and 556. We should take a look at resolving those.

**Comment by gailweiss** [comment]
this is a very big change (touches all of the files), and anyway, i rather like tabs for indentation :) so i'm avoiding this one. That said, I won't object to receiving a separate PR for removing unused imports!

**Comment by emileferreira** [comment]
Yes it is a very big change, which required a lot of time and effort, but I have ensured that all tests pass. A well-formatted codebase would make it easier for others to contribute to the project. I have now reverted the files to tab-based indentation and hope that you consider re-opening the pull request.

**Comment by gailweiss** [comment]
Thanks! Can you please move the is/== change you made in Sugar.py to a separate PR, as it touches the code itself and I would like to handle it separately. Also, have you committed this with —ignore-rev, and if not can you add it (https://akrabat.com/ignoring-revisions-with-git-blame/)?

**Comment by emileferreira** [comment]
I recall making such a change, but I don't see it in `Sugar.py`. Could you please point me to the line number? I have granted you write access to the pull request branch, should you wish to make any changes.

The `--ignore-rev` argument can't be used when making commits; it's only for use with `git blame` to locally ignore commits. Once you are satisfied with the changes, you can squash the commits to make you the author or I can edit the author details.

**Comment by emileferreira** [comment]
I've reverted the offending commit and created a `.gitignore-revs` file. As per the linked article, you need to point to this file from your local Git config to automatically ignore the specified commits.

**Comment by emileferreira** [comment]
I now realise why I couldn't see the change in `Sugar.py`; the original file used the `==` operator. I changed it to an `is` operator in 309fb81 before restoring it in 31ca5b2. I have reverted `Sugar.py` to the original operator.

**Comment by gailweiss** [comment]
i went over it. had to make a few changes -- mostly fixing broken comments, but also some variable names -- and made a PR to your PR, i think. thanks for the thorough work, looking forward to merge once you merge in my changes

**Comment by emileferreira** [comment]
Thanks for your changes—they've been merged. I've reverted your reversion of the ambiguous variable names, as discussed and in compliance with PEP 8's [names to avoid](https://peps.python.org/pep-0008/#names-to-avoid).

File: RASP_support/Support.py [hunk 1]
```diff
@@ -176,15 +176,17 @@ def zipmap(n, k_vars, func):
 def verify_default_size(default, num_output_vars):
 	assert num_output_vars > 0
 	if num_output_vars == 1:
-		assert not isinstance(
-			default, tuple), "aggregates on functions with single output \
-			should have scalar default"
+		errnote = "aggregates on functions with single output should have" \
+			+ "scalar default"
```
Comment (gailweiss) [inline RASP_support/Support.py:180]: oops, can you add a space here to sit between "have" and "scalar" in the string?

File: RASP_support/Evaluator.py [hunk 2]
```diff
@@ -224,7 +224,7 @@ def _evaluateDictComp(self, ast):
 		if not (isinstance(d, list) or isinstance(d, dict)):
 			raise RASPTypeError(
 				"dict comprehension should have got a list or dict to loop "
-				+ "over, but got:", l)
+				+ "over, but got:", d)
```
Comment (gailweiss) [inline RASP_support/Evaluator.py:227]: good catch but i actually was hoping to get this in a separate pr cause it fixes an actual bug

File: RASP_support/Evaluator.py [hunk 3]
```diff
@@ -500,7 +500,7 @@ def _evaluateAggregateExpr(self, ast):
 		default = self.evaluateExpr(ast.default) if ast.default else None
 
 		if not isinstance(sel, UnfinishedSelect):
-			raise RASPTypeError("Expected selector, got:", strdesc(selector))
+			raise RASPTypeError("Expected selector, got:", strdesc(sel))
```
Comment (gailweiss) [inline RASP_support/Evaluator.py:503]: please put in separate bug pr (not a pep 8 fix)

File: RASP_support/Evaluator.py [hunk 4]
```diff
@@ -552,7 +552,7 @@ def _evaluateApplication(self, ast, unf):
 		if not isinstance(unf, Unfinished):
 			raise RASPTypeError("Applying unfinished expects to apply",
 								ENCODER_NAME, "or selector, got:",
-								strdesc(sel))
+								strdesc(unf))
```
Comment (gailweiss) [inline RASP_support/Evaluator.py:555]: please put in separate bug pr (not a pep 8 fix)

**Review by gailweiss — CHANGES_REQUESTED** [review]
you fixed some bugs, which i appreciate but would prefer in a separate pr

**Comment by gailweiss** [comment]
make sure to add the latest commits to the git blame ignore file (although maybe we'll squash and ill have to update it, ill see what happens - terrible at this)

File: RASP_support/Support.py (hunk 1 above)
Comment (emileferreira) [reply RASP_support/Support.py]: No problem :)

**Pull request merged**